pobi.md

```diff
@@ -54,74 +54,73 @@ Note that the value of L1_Proof_Height is less than L1_Block_Height.
+The POBI protocol allows any aggregator to publish rollups to the management contract, so short-lived forks are a normal part of the protocol. However, the ObscuroVM running inside the TEE of every node must be able to deterministically select one of the forks as the canonical chain and only append a rollup on top of that. This means that the TEE must receive all the relevant content of the L1 blocks, and the logic must be identical on all nodes.
 
 The rules for the canonical chain:
-1. The genesis rollup is part of the canonical chain, and will be included in a block by the first aggregator.
-2. If an L1 block contains a single rollup whose parent is the head rollup of the canonical chain included in a previous L1 block, it will also be on the canonical chain if no other rollup with the same parent was included in an earlier block. Any other sibling rollup included in a later block is not on the canonical chain. This is the _Primogeniture_ rule, where a rollup is born when included in an L1 block.
-3. If an L1 block contains multiple sibling rollups created in the same round using the same L1 proof, the one with the lower nonce will be on the canonical chain.
-4. If an L1 block contains multiple sibling rollups created using different L1 proofs, the one created more recently is on the canonical chain.
+1. The genesis rollup is part of the canonical chain and will be included in a block by the first aggregator.
+2. A L1 block containing a single rollup whose parent is the head rollup of the canonical chain included in a previous L1 block is on the canonical chain if no other rollup with the same parent was included in an earlier block. Any other sibling rollup included in a later block is not on the canonical chain. This is the _Primogeniture_ rule, where a rollup is born when included in a L1 block.
+3. If a L1 block contains multiple sibling rollups created in the same round using the same L1 proof, the one with the lower nonce, will be on the canonical chain.
+4. If a L1 block contains multiple sibling rollups created using different L1 proofs, the one created more recently is on the canonical chain.
 
 [comment]: <> ([TODO - diagram depicting these scenarios])
 
-Using the notation, for the same _Rollup_Generation_, the rollup that will be on the canonical chain is the one with:
+Using the notation, for the same _Rollup_Generation_, the rollup on the canonical chain is the one with:
 1. The lowest L1_Block_Generation. 
 2. In case there are multiple matches, use the highest L1_Proof_Generation. 
 3. In case there are multiple matches, use the lowest nonce.
 
-Given that the nonce is a random number with sufficient entropy, we assume there cannot be a collision at this point.
```

Comment:
the nonce

Reply:
oops, now that I look at this, there might be a problem in case of a tee hack. the attacker could create equal nonces